src/parser/jobs/rdm/modules/ManaStackGauge.tsx

```diff
@@ -0,0 +1,214 @@
+	timestamp: number,
+	manaStacksLost: number,
+}
+export const MINIMUM = 0
+export const MAXIMUM = 3
+
+export class ManaStackGauge extends CoreGauge {
+	static override handle = 'manaStackGauge'
+	static override title = t('rdm.manaStackGauge.title')`Mana Stack Gauge Usage`
+	static override debug = false
+
+	@dependency private suggestions!: Suggestions
+	@dependency private statistics!: Statistics
+	@dependency private gauge!: Gauge
+
+	private manaStackGauge = this.add(new CounterGauge({
+		graph: {
+			label: <Trans id="rdm.manaStackGauge.resource.manaStacks">Mana Stacks</Trans>,
+			color: JOBS.RED_MAGE.colour,
+		},
+		maximum: MAXIMUM,
+		minimum: MINIMUM,
+	}))
+	public gaugeModifiers = new Map<number, ManaStackGaugeModifier>([
+		[this.data.actions.ENCHANTED_REPRISE.id, {manaStack: 1}],
+		[this.data.actions.ENCHANTED_MOULINET.id, {manaStack: 1}],
+		[this.data.actions.ENCHANTED_RIPOSTE.id, {manaStack: 1}],
+		[this.data.actions.ENCHANTED_ZWERCHHAU.id, {manaStack: 1}],
+		[this.data.actions.ENCHANTED_REDOUBLEMENT.id, {manaStack: 1}],
+	])
+	public spenderModifiers = new Map<number, ManaStackGaugeModifier>([
+		[this.data.actions.VERHOLY.id, {manaStack: -1}],
+		[this.data.actions.VERFLARE.id, {manaStack: -1}],
+		[this.data.actions.SCORCH.id, {manaStack: -1}],
+		[this.data.actions.RESOLUTION.id, {manaStack: -1}],
+	])
+	public gaugeBreakers: number[] = [
+		this.data.actions.VERAERO.id,
+		this.data.actions.VERAERO_II.id,
+		this.data.actions.VERAERO_III.id,
+		this.data.actions.VERSTONE.id,
+		this.data.actions.VERTHUNDER.id,
+		this.data.actions.VERTHUNDER_II.id,
+		this.data.actions.VERTHUNDER_III.id,
+		this.data.actions.VERFIRE.id,
+		this.data.actions.JOLT.id,
+		this.data.actions.JOLT_II.id,
+		this.data.actions.IMPACT.id,
+	]
+	gaugeBreak: GaugeBreaker[] = []
+	//Any loss is a major severity
+	severity = {
+		1: SEVERITY.MAJOR,
+	}
+	manaStatistics = {
+		white: {
+			overcapLoss: 0,
+		},
+		black: {
+			overcapLoss: 0,
+		},
+	}
+
+	override initialise() {
+		super.initialise()
+
+		this.addEventHook(
+			filter<Event>()
+				.type('action')
+				.source(this.parser.actor.id)
+				.action(oneOf(Array.from(this.gaugeModifiers.keys()))),
+			this.onGaugeModifying
+		)
+		this.addEventHook(
+			filter<Event>()
+				.type('action')
+				.source(this.parser.actor.id)
+				.action(oneOf(this.gaugeBreakers)),
+			this.onGaugeModifying
+		)
+		this.addEventHook(
+			filter<Event>()
+				.type('action')
+				.source(this.parser.actor.id)
+				.action(oneOf(Array.from(this.spenderModifiers.keys()))),
+			this.onGaugeSpender
+		)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onGaugeModifying(event: Events['action']) {
+		const modifier = this.gaugeModifiers.get(event.action)
+
+		//Always check it we should break
+		if (this.onGaugeBreak(event)) {
+			//We won't modify the gauge if this skill broke it.
+			return
+		}
+
+		if (modifier == null) {
+			return
+		}
+
+		if (this.manaStackGauge.capped) {
```

Comment:
I feel like this whole block should be removed. We aren't actually losing white/black mana by hitting an extra mana gauge builder.

Reply:
How is it not lost if I spent 70|70 for a finisher that should only cost 50|50?  That 20|20 isn't going to apply to my next Mana Stack Build, it's not mana I'm gonna get back, it's a total loss of 20|20 mana.

Reply:
I suppose I understand. "Lost" is a strong word I guess, its not like you didn't do ANY damage with the mana (which is the case for Using Manafication over 50/50 e.g.). I'm fine with it and The Balance folks can decide if they like it or not.

Reply:
I do kind of agree that this function is confusing, and the first debug line's copy doesn't help the confusion either.  

Plus, this isn't going to generate the correct amount of overcap if the user does Moulinet > Riposte > Zwerchau > Redoublement - this is going to look up the 15|15 of Redoublement and add that to overcap, even though the user spent 70|70 (same scenario applies if user starts with 2x Riposte).

It seems to me that if we want to track overspend loss in the White/Black gauge, we're going to need the Mana Stack gauge to track state of each "group" of builders, and then evaluate those builders against what was necessary in order to get a Verholflare combo.

Reply:
In the case of the example you gave, the correct answer is 15|15 lost.  The fact that potency was lost due to the bad management of enchanted skills is immaterial.  Any enchanted skill will generate a Mana Stack, no matter the order utilized except for Reprise so long as no other GCD skill was utilized between them.  This metric is only concerned with specifying the mana that was purely wasted on a mana stack over the 3 cap.  If I truly wanted to, I could do 3x Redoublement to go into the finisher combo, it would just be a horrible potency choice.

As for that debug line, that was from early in development and I just forgot to delete it once I organized the code better is all.